docs/guides/custom-functions.md

```diff
@@ -2,6 +2,10 @@
 
 If the built-in functions are not enough for your [custom ruleset](../getting-started/rulesets.md), Spectral allows you to write and use your own custom functions.
 
+As of Spectral 5.4.0, custom functions can also be asynchronous. 
+Be aware that linting should be deterministic. That being said, please refrain from introducing any logic that is prone to non-deterministic behavior, i.e. contacting external service you have no control of, and hence cannot predict the response.
```

Comment:
This could be a warning block.

```suggestion
Ideally linting should always be deterministic, which means if its run 10 times it should return the same results 10 times. To ensure this is the case, please refrain from introducing any logic that is prone to non-deterministic behavior. Examples of this might be contacting external service you have no control over, or that might be unstable, or change the way it responds over time.
```

Reply:
> This could be a warning block.

Do you mean the warning block from SMD?